src/database.py

```diff
@@ -0,0 +1,147 @@
+
+        Raises:
+           Error: something went wrong while preparing the database.
+        """
+        conn = self.get_conn()
+        database_name = self.get_database_name()
+        if conn is not None:
+            try:
+                with conn.cursor() as curs:
+                    curs.execute(
+                        sql.SQL(
+                            "UPDATE pg_database "
+                            "SET datcollate='C', datctype='C' "
+                            "WHERE datname = {}"
+                        ).format(sql.Literal(database_name))
+                    )
+            except psycopg2.Error as exc:
+                logger.error("Failed to prepare database: %s", str(exc))
+                raise
+
+    def erase_database(self) -> None:
+        """Erase database.
+
+        Raises:
+           Error: something went wrong while erasing the database.
+        """
+        # Since is not possible to delete the database while connected to it
+        # this connection will use the template1 database, provided by PostgreSQL.
+        conn = self.get_conn("template1")
+        database_name = self.get_database_name()
+        if conn is not None:
+            try:
+                with conn.cursor() as curs:
+                    curs.execute(sql.SQL("DROP DATABASE {}").format(sql.Identifier(database_name)))
+                    curs.execute(
+                        sql.SQL(
+                            "CREATE DATABASE {} "
+                            "WITH LC_CTYPE = 'C' LC_COLLATE='C' TEMPLATE='template0';"
+                        ).format(sql.Identifier(database_name))
+                    )
+            except psycopg2.Error as exc:
+                logger.error("Failed to erase database: %s", str(exc))
+                raise
+
+    def get_conn(self, database_name: str = "") -> connection | None:
+        """Get connection.
+
+        Args:
+            database_name: database to connect to.
+
+        Raises:
+           Error: something went wrong while connecting to the database.
+
+        Returns:
+            Connection or None if there is no relation data
+        """
+        relation_data = self.get_relation_data()
+        if relation_data is not None:
+            try:
+                user = relation_data.get("POSTGRES_USER")
+                password = relation_data.get("POSTGRES_PASSWORD")
+                host = relation_data.get("POSTGRES_HOST")
+                if not database_name:
+                    database_name = self.get_database_name()
+                conn = psycopg2.connect(
+                    f"dbname='{database_name}' user='{user}' host='{host}'"
+                    f"password='{password}' connect_timeout=1"
+                )
+                conn.autocommit = True
+                return conn
+            except psycopg2.Error as exc:
+                logger.error("Failed to connect to database: %s", str(exc))
+                raise
```

Comment:
I think `logger.exception` is better in this scenario, since it will print the trackback and full exception information.

Reply:
Nice, thanks!